generator-test/src/test/java/com/linkedin/pegasus/generator/override/TestRecord.java

```diff
@@ -530,4 +530,15 @@ public void testCopyChangeRecordField() throws CloneNotSupportedException
     assertEquals(record.getRecordOptionalField().getLocation(), "near");
     assertNull(recordCopy.getRecordOptionalField());
   }
+
+  @Test
+  public void testSetOnRecordWrappingSameMap()
+  {
+    RecordBar bar = new RecordBar();
+    bar.setLocation("some");
+    RecordBar copy = new RecordBar(bar.data());
```

Comment:
What about deep copy (if we have any)

Reply:
Those are already tested via other test cases.